content/en/tutorials/gitlab_ci_testcontainers/index.md

```diff
@@ -0,0 +1,431 @@
+
+## GitLab architecture
+
+{{< figure src="gitlab-architecture.png" width="80%" height="auto">}}
+
+<br>
+
+As users, we only interact directly with a GitLab instance which is responsible for hosting the application code and all the needed configurations, including the 
+ones for pipelines. The instance is then in charge of running the pipelines and assigning runners to execute the defined jobs.
+
+When running CI pipelines, you can choose to use [**GitLab-hosted runners**](https://docs.gitlab.com/ee/ci/runners/index.html), or provision and register 
+[**self-managed runners**](https://docs.gitlab.com/runner/install/docker.html). This tutorial will cover both.
+
+#### Runners hosted by GitLab
+
+The GitLab documentation highlights some key aspects about the provided runners:
+
+- They can run on Linux, Windows (beta) and MacOS (beta).
+- They are enabled by default for all projects, with no configuration required.
+- Each job is executed by a newly provisioned VM. 
+- Job runs have `sudo` access without a password.
+- VMs are isolated between job executions. 
+- Their storage is shared by the operating system, the image with pre-installed software,
+and a copy of your cloned repository, meaning that the remaining disk space for jobs will be reduced.
+- The runners are configured to run in privileged mode to support Docker in Docker to build images natively or 
+run multiple containers within each job.
+
+#### Self-hosted runners
+
+Essentially, the architecture does not change, except the runners will be executing the jobs on a local machine. For developing locally, 
+this approach is very convenient and there are several benefits:
+
+- **Customization**: you can configure the runners to suit your specific needs and environment. 
+- **Performance**: improved performance and faster builds by leveraging your own hardware. 
+- **Security**: enhanced control over your data and build environment, reducing exposure to external threats. 
+- **Resource Management**: better management and allocation of resources to meet your project's demands.
+- **Cost Efficiency**: depending on your alternatives, you can avoid usage fees associated with cloud-hosted runners.
+
+
+## Application Overview
+
+Our sample backend application stores information about different types of coffee in files, with descriptions stored in an S3 bucket. It utilizes two
+Lambda functions to create/update and retrieve these descriptions, all accessible through an API Gateway. While we won't delve
+into the details of creating these AWS resources, we'll use AWS CLI to initialize them during container startup using init hooks. You can
+find the whole setup in the [init-resources.sh](https://gitlab.com/tinyg210/coffee-backend-localstack/-/blob/main/src/test/resources/init-resources.sh?ref_type=heads) file.
```

Comment:
Please migrate this to https://gitlab.com/localstack.cloud, since maintaining repos on personal accounts/namespaces is not a good precedent

Reply:
@whummer Can you help @tinyg210 with access to the GitLab repo?

Reply:
i'll swap links when i get access to it